src/compiler/compiler.rs

```diff
@@ -1401,10 +1401,7 @@ LLVM version: 6.0",
             }))
             .unwrap();
         // Ensure that the object file was created.
-        assert_eq!(
-            true,
-            fs::metadata(&obj).and_then(|m| Ok(m.len() > 0)).unwrap()
-        );
+        assert_eq!(true, fs::metadata(&obj).map(|m| m.len() > 0).unwrap());
```

Comment:
Should this be `assert!(...)`, instead of asserting that `true` is equal to your computation?
This is old code, but hey, while we're fixing lint-worthy issues :)